x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/action/EqlSearchResponse.java

```diff
@@ -263,6 +266,11 @@ public Event(StreamInput in) throws IOException {
             }
         }
 
+        public static Event readFrom(StreamInput in) throws IOException {
+            Event result = new Event(in);
+            return result.equals(MISSING) ? MISSING : result;
```

Comment:
Would a boolean as null indicator work as well, similar to the serialisation of other ["optionals"](server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java)? Though having a marker object works too.

Reply:
The problem is actually in serialization: the events are in a collection nested in the EQL response, and `StreamOutput.writeCollection()` cannot handle nulls, so a local fix (eg. a boolean in the serialization) is not enough

Reply:
I was wondering if the "local" serialisation couldn't deal with a boolean flag, rather than de-/serialise an Event-as-null. But yes, it might require version-dependent behaviour, if this had worked at all before.